aws/rust-runtime/aws-sigv4/src/http_request/canonical_request.rs

```diff
@@ -317,7 +317,12 @@ impl<'a> fmt::Display for CanonicalRequest<'a> {
             // a missing header is a bug, so we should panic.
             let value = &self.headers[&header.0];
             write!(f, "{}:", header.0.as_str())?;
-            writeln!(f, "{}", value.to_str().unwrap())?;
+            writeln!(
```

Comment:
I wonder if we should just use the header value's bytes in the canonical request. Although, doing so would require some refactoring of the sigv4 crate since it wouldn't be able to use the `Display` trait anymore for formatting this since the bytes could be invalid UTF-8.

Reply:
I don't think it's possible from smithy to ever put non-utf8 encoded data into a header value, IIRC. So this should be enough.